app/queries/with_one_matching_value_for_each_key.rb

```diff
@@ -0,0 +1,45 @@
+class WithOneMatchingValueForEachKey
```

Comment:
It's late, and I might be misunderstanding this, but `WithAtLeastOneMatchingValue` is more accurate, I think.